### ISSUE TITLE
Option to indent code in a #region 

### ISSUE THREAD

**Issue by WillyMoselhy** [PowerShell/vscode-powershell#3972]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.

### Summary

When writing code, I depend heavily on regions to keep things clean. Sometimes even nested regions. I would love it if we have the option to indent the code in a region to easily find where a region starts and ends.
```PowerShell
#region: Hello World!
Write-Host 'Hello World!
#endregion
```
Should be (optinally)
```PowerShell
#region: Hello World!
    Write-Host 'Hello World!
#endregion
```
Would be even cooler if the indent is adjustable to 1 or 2 spaces instead of 4

### Proposed Design

Add option in the formatter to indent regions and specifying by how many spaces.

**Comment by bergmeister** [comment]
Yes, although regions are not part of the AST and not a Token either so writing a ScriptAnalyzer rule will be a bit hacky to parse the regions out, it shouldn't be too bad though so please transfer the issue to PSSA repo
